File: 2089-find-target-indices-after-sorting-array/2089-find-target-indices-after-sorting-array.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var targetIndices = function(nums, target) {
  const ans = []
  nums.sort((a, b) => a - b).forEach((n, i) => (n === target) ? ans.push(i) : 0)
  return ans
};

File: 2089-find-target-indices-after-sorting-array/2089-find-target-indices-after-sorting-array.go
func targetIndices(nums []int, target int) []int {
  sort.Ints(nums)
  l, r := 0, len(nums) - 1
  for l <= r {
    mid := (l + r) / 2
    if nums[mid] < target {
      l = mid + 1
    } else {
      r = mid - 1
    }
  }
  left := l
  l, r = 0, len(nums) - 1
  for l <= r {
    mid := (l+r) / 2
    if nums[mid] > target {
      r = mid - 1
    } else {
      l = mid + 1
    }
  }
  if left > r {
    return []int{}
  } else {
    ans := make([]int, 0, r + 1 - left)
    for i := left; i <= r; i++ {
      ans = append(ans, i)
    }
    return ans
  }
}